lib/command.js

```diff
@@ -223,6 +223,24 @@ module.exports = function command (yargs, usage, validation, globalMiddleware) {
     if (!yargs._hasOutput()) {
       positionalMap = populatePositionals(commandHandler, innerArgv, currentContext, yargs)
     }
+    let preCheckMiddlewares = globalMiddleware.slice(0).concat(commandHandler.middlewares || []).filter((x) => x.applyBeforeValidation === true)
```

Comment:
similarly to how we pulled `applyMiddleware` into the `middleware` module, I think for consistency this should be pulled into `middleware.js` as `applyPreCheckMiddleware`; Let's try to encapsulate as much logic specific to middleware as possible in its own helper file (`command.js` is pretty unwieldy).

Reply:
I've moved the logic from command.js into middleware.jjs